src/main/java/com/example/sixthmafiabot/validators/UserValidator.java

```diff
@@ -16,71 +18,70 @@
 import java.util.concurrent.CompletableFuture;
 
 @Component
+@Slf4j
 public class UserValidator extends BaseValidator {
 
     @Autowired
     UserRepository userRepository;
 
     @Async("asyncExecutor")
-    public void validateUser(User user) throws ValidationException{
-        Set<ConstraintViolation<User>> violations =
-                validator.validate(user);
-
-        if (!violations.isEmpty()) {
-            StringBuilder sb = new StringBuilder();
-            for (ConstraintViolation<User> constraintViolation : violations) {
-                sb.append(constraintViolation.getMessage());
-            }
-            throw new ValidationException("Error occurred: " + sb.toString());
-        }
+    public void validateCreateUser(CreateUserDTO user) throws ValidateException{
+
+        validateDTO(user);
+    }
+
+    public void validateUpdateUser(UpdateUserDTO user) throws ValidateException{
+
+        validateDTO(user);
     }
 
     @Async("asyncExecutor")
-    public void validateIfAlreadyExists(User user) throws AlreadyExistsExcepeion {
+    public void validateIfAlreadyExists(CreateUserDTO user) throws AlreadyExistsExcepeion {
 
         boolean alreadyExists =  userRepository
                 .getUserByTelegramId(user.getTelegramId())
                 .join() != null;
 
         if(alreadyExists){
+
             throw new AlreadyExistsExcepeion("User with identifier = %s already exists"
                     .formatted(user.getTelegramId())
             );
         }
-
     }
 
     @Async("asyncExecutor")
-    public CompletableFuture<User> validateCreate(User user) throws ServiceValidationError {
+    public CompletableFuture<CreateUserDTO> validateCreate(CreateUserDTO createUserDTO) throws ServiceValidationError {
```

Comment:
why do we have both validateCreate and validateCreateUser

Reply:
validateCreate uses validateCreateUser and other methods, we call validateCreate from service, validateCreateUser validates only labels (annotations)